### PR TITLE
apis implementation for classroom and attendance routes

### PR DESCRIPTION
**Pull request by shradhaj06**
PR Description:
This PR introduces changes to the backend API routes for managing classroom and attendance records. The following updates have been made:

1. *Classroom Routes*:
   - Implemented CRUD (Create, Read, Update, Delete) operations for managing classroom records.
   - Created routes for creating, retrieving all, retrieving by ID, updating, and deleting classroom records.
   - Implemented controller functions to handle each route operation.
   - Added error handling for handling invalid requests and internal server errors.

2. *Attendance Routes*:
   - Implemented CRUD operations for managing attendance records.
   - Created routes for creating, retrieving all, retrieving by ID, updating, and deleting attendance records.
   - Developed controller functions to handle each route operation.
   - Included error handling mechanisms to manage errors gracefully.

3. *Controller Functions*:
   - Defined controller functions for each route operation to ensure separation of concerns and maintainability.
   - Created functions for getting all records, getting records by ID, creating new records, updating records, and deleting records.

4. *Testing*:
   - Tested each route to ensure proper functionality and error handling.

These changes enhance the backend API's functionality and improve its robustness by adding support for managing classroom and attendance records effectively. The PR aims to enhance the overall performance and reliability of the backend system.

### CODE REVIEW

File: ams_flutter/lib/features/onboarding/presentation/User/Pages/providedetailspage.dart
Comment (RitwikSingh28) [inline ams_flutter/lib/features/onboarding/presentation/User/Pages/providedetailspage.dart:1]: @shradhaj06 
How is this file relevant to your PR?

**Review by RitwikSingh28 — APPROVED** [review]
@shradhaj06 
Great work! I couldn't find any flaws with the code written.

File: ams_flutter/lib/features/onboarding/presentation/User/Pages/providedetailspage.dart
Comment (shradhaj06) [reply ams_flutter/lib/features/onboarding/presentation/User/Pages/providedetailspage.dart:1]: It was showing some import error, so I just fixed it. Nothing substantial done there.
